dev-python/pyroute2/pyroute2-0.5.19.ebuild

```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python3_{7,8,9,10} pypy3 )
+PYTHON_REQ_USE="sqlite"
+DISTUTILS_USE_SETUPTOOLS=bdepend
+
+inherit distutils-r1
+
+DESCRIPTION="A pure Python netlink and Linux network configuration library"
+HOMEPAGE="https://github.com/svinota/pyroute2"
+SRC_URI="https://github.com/svinota/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 x86"
+IUSE="test"
+
+BDEPEND="test? ( dev-python/psutil[${PYTHON_USEDEP}] )"
+
+RESTRICT="!test? ( test )"
+
+# tests need root access
+RESTRICT+=" test"
```

Comment:
Could you at least get a subset of them to work?

Reply:
i tried looking into the tests but getting them run, maybe partially, seems like a task for the move to 0.6 or even an improvement on top

we have @svinota from upstream also wanting to support gentoo

i tried that 0.5.19 testsuite and it got stuck as non-root in a very early stage checking capabilities, maybe the 0.6 testsuite can skip bits on its own when running as non-root. Enabling the tests as non-root is a discussion to have with upstream, and only against their most recent version.

So i removed all testing bits and only placed that `RESTRICT` for now

Reply:
upstream currently also simply says that one needs root https://github.com/svinota/pyroute2/blob/5036076750bb4efa7dad318ad49a1ba69e6ca01f/README.make.md?plain=1#L22

digging in there and patching around to get it partially working without upstream supporting that effort ... not a good idea